execution/state.go

```diff
@@ -172,6 +181,84 @@ func MakeGenesisState(db dbm.DB, genesisDoc *genesis.GenesisDoc) (*State, error)
 	return s, nil
 }
 
+func (s *State) LoadDump(filename string, genesisDoc *genesis.GenesisDoc) error {
+	f, err := os.OpenFile(filename, os.O_RDONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	tx := exec.TxExecution{
+		TxType: payload.TypeCall,
+		TxHash: make([]byte, 32),
+	}
+
+	for {
+		var row dump.Dump
+
+		_, err = cdc.UnmarshalBinaryLengthPrefixedReader(f, &row, 0)
+		if err != nil {
+			break
+		}
+
+		if row.Account != nil {
+			if row.Account.Address == crypto.ZeroAddress {
+				if genesisDoc != nil {
+					genesisDoc.GlobalPermissions.Base = row.Account.Permissions.Base
+				}
+				continue
+			}
+
+			s.writeState.UpdateAccount(row.Account)
+
+			if genesisDoc != nil {
+				found := false
+
+				if len(row.Account.PublicKey.PublicKey) == 0 {
+					// no public key, so not an account
+					continue
+				}
+
+				for _, acc := range genesisDoc.Accounts {
+					if acc.Address == row.Account.Address {
+						found = true
+						break
+					}
+				}
+				if !found {
+					genesisDoc.Accounts = append(genesisDoc.Accounts, genesis.GenesisAccountFromAccount("", row.Account))
```

Comment:
Hmmm... We are saying here that _every_ account is a genesis account?

The problem with this is that we break the idea that the genesis doc on disk matches the genesis doc in memory. Looking at how you have used this however I don't think it is broken (since you pass nil in kernel)...

So the question is do we want configure to create all accounts in a dump as 'genesis accounts'. This feels a bit wrong to me since providing AppHash should mean give me all of what's in the dump plus I want these explicit accounts. I can see both ways...

Reply:
Removed